File: 001/pseudoinverse.jl
function run()
    y = [0; 1; 4; 9]
    data = [0 1; 1 1; 2 1; 3 1]
    trans = transpose(data)
    square = trans * data
    inverse = inv(square)
    answer = inverse * trans * y
end

start = time_ns()
for i in 1:10
    run()
end
stop = time_ns()
print("Julia: ", (stop-start)/(10^10))

File: 001/pseudoinverse.py
import numpy as np
import time
print("\npsuedoinverse")
def run():
    y = [[0],[1],[4],[9]]
    data = [[0,1],[1,1],[2,1],[3,1]]
    transpose = np.transpose(data)
    square = np.matmul(transpose, data)
    inverse = np.linalg.inv(square)
    answer = np.matmul(inverse, np.matmul(transpose, y))

start = time.time()
for _ in range(10):
    run()
end = time.time()
print("Python: ", (end-start)/10)

